FreeRTOS_ARP.c

```diff
@@ -152,8 +245,12 @@ eFrameProcessingResult_t eARPProcessPacket( ARPPacket_t * const pxARPFrame )
                 case ipARP_REQUEST:
 
                     /* The packet contained an ARP request.  Was it for the IP
-                     * address of the node running this code? */
-                    if( ulTargetProtocolAddress == *ipLOCAL_IP_ADDRESS_POINTER )
+                     * address of the node running this code? And does the MAC
+                     * address claim that it is coming from this device itself? */
+                    if( ( ulTargetProtocolAddress == *ipLOCAL_IP_ADDRESS_POINTER ) &&
+                        ( memcmp( ( void * ) ipLOCAL_MAC_ADDRESS,
+                                  ( void * ) ( pxARPHeader->xSenderHardwareAddress.ucBytes ),
+                                  sizeof( MACAddress_t ) ) != 0 ) )
```

Comment:
It's an interesting case because `ucBytes` is the only member in the struct so it's unlikely to have [alignment issues](https://stackoverflow.com/questions/3585033/size-of-struct-with-a-single-element). However, it is not necessarily guaranteed, so feel like it would be best to use `sizeof(pxARPHeader->xSenderHardwareAddress.ucBytes)` instead.

Reply:
In most cases in the library, `ipMAC_ADDRESS_LENGTH_BYTES` is used, which is defined as 6.
All struct's that describe network data are packed, so `MACAddress_t` and `MACAddress_t::ucBytes` are both 6 bytes long.
When some compiler doesn't pack the struct's, the library won't work at all.

Reply:
Yes, as Hein said, if a compiler doesn't pack the structures - the whole library will not work.
Also, I shall modify the number of bytes to `ipMAC_ADDRESS_LENGTH_BYTES`.